src/monkey_patch/language_models/bedrock_api.py

```diff
@@ -0,0 +1,124 @@
+import abc
+import json
+
+import boto3 as boto3
+import openai
+import time
+# import abstract base class
+from monkey_patch.language_models.llm_api_abc import LLM_Api
+import os
+
+OPENAI_URL = "https://api.openai.com/v1/chat/completions"
+import requests
+
+AWS_REGION = os.environ.get("AWS_DEFAULT_REGION", "us-east-1")
+bedrock = boto3.client(
+    service_name='bedrock',
+    region_name=AWS_REGION,
+)
+bedrock_runtime = boto3.client(
+    service_name='bedrock-runtime',
+    region_name=AWS_REGION,
+
+)
+
+
+class PromptTemplate:
+
+    @abc.abstractmethod
+    def prompt_gen(self, system_prompt, user_prompt):
+        pass
+
+
+class AnthropicClaudePromptTemplate(PromptTemplate):
+
+    def prompt_gen(self, system_prompt, user_prompt):
+        return f"\n\nHuman: {system_prompt}\n\n {user_prompt}\n\nAssistant:\n"
+
+
+class Bedrock_API(LLM_Api):
+    def __init__(self) -> None:
+        # initialise the abstract base class
+        super().__init__()
+
+    def generate(self, model, system_message, prompt, **kwargs):
+        """
+        The main generation function, given the args, kwargs, function_modeler, function description and model type, generate a response and check if the datapoint can be saved to the finetune dataset
+        """
+
+        # check if api key is not none
+        content_type = 'application/json'
+
+        temperature = kwargs.get("temperature", 0)
+        top_p = kwargs.get("top_p", 1)
+        frequency_penalty = kwargs.get("frequency_penalty", 0)
+        presence_penalty = kwargs.get("presence_penalty", 0)
+        body = json.dumps({
+            "prompt": AnthropicClaudePromptTemplate().prompt_gen(system_prompt=system_message, user_prompt=prompt),
+            "max_tokens_to_sample": 4096,
+            "temperature": temperature,
+            "top_k": 250,
+            "top_p": 1,
+            "stop_sequences": [
+                "\\n\\nHuman:"
+            ],
+            "anthropic_version": "bedrock-2023-05-31"
+        })
+
+        response = bedrock_runtime.invoke_model(
+            body=body,
+            modelId=model,
+            contentType="application/json",
+            accept="application/json")
+        return json.loads(response.get('body').read().decode())['completion']
```

Comment:
We shouldn't be returning this here - we should loop the counter below in the case where the `bedrock_runtime` throws an exception.